aggregator/src/aggregator.rs

```diff
@@ -1761,358 +1755,318 @@ impl VdafOps {
             }
         }
 
-        let prep_closure = {
-            let parent_span = Span::current();
-            let global_hpke_keypairs = global_hpke_keypairs.view();
-            let vdaf = Arc::clone(&vdaf);
-            let task = Arc::clone(&task);
-            let metrics = metrics.clone();
-            let req = Arc::clone(&req);
-            let aggregation_job_id = *aggregation_job_id;
-            let verify_key = *verify_key;
-            let agg_param = Arc::clone(&agg_param);
-            move || -> Result<Vec<ReportShareData<SEED_SIZE, A>>, Error> {
-                let span = info_span!(
-                    parent: parent_span,
-                    "handle_aggregate_init_generic threadpool task"
+        // We validate that each prepare_init can be represented by a `u64` ord value here, so that
+        // inside the parallel iterator we can unwrap. A conversion failure here will fail the
+        // entire aggregation. However, this is desirable: this can only happen if we receive too
+        // many report shares in an aggregation job for us to store, which is a whole-aggregation
+        // problem rather than a per-report problem. (separately, this would require more than
+        // u64::MAX report shares in a single aggregation job, which is practically impossible.)
+        u64::try_from(req.prepare_inits().len())?;
+
+        let mut report_share_data = Vec::new();
```

Comment:
nit: We can pre-allocate this vector

Reply:
Fixed by merging the follow-on PR (lol).